java/core-arrow/src/main/java/sleeper/arrow/schema/SchemaConverter.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.arrow.schema;
+
+import sleeper.core.schema.Field;
+import sleeper.core.schema.Schema;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SchemaConverter {
```

Comment:
The name could be more specific that this is for Arrow, maybe ArrowSchemaConverter? The same goes for FieldConverter.